### PR TITLE
fixed spelling errors in whatsnew, userguide, and ecosystem

### PR DESCRIPTION
**Pull request by asharma13524**
Hi there, 

I fixed the spelling errors mentioned here. https://github.com/pandas-dev/pandas/issues/37252

 The 2 "wit" spelling errors were confusing, and I could not understand the sentences. 

This is my first pull request, so please excuse my ignorance if I did something incorrectly.

Thanks.

### CODE REVIEW

File: doc/source/user_guide/indexing.rst [hunk 1]
```diff
@@ -584,7 +584,7 @@ IX indexer is deprecated
 
    The ``.ix`` indexer was removed, in favor of the more strict ``.iloc`` and ``.loc`` indexers.
 
-``.ix`` offers a lot of magic on the inference of what the user wants to do. To wit, ``.ix`` can decide
```
Comment (dsaxton) [inline doc/source/user_guide/indexing.rst:587]: I think "To wit" is correct

File: doc/source/whatsnew/v0.20.0.rst [hunk 2]
```diff
@@ -1308,7 +1308,7 @@ Deprecations
 Deprecate ``.ix``
 ^^^^^^^^^^^^^^^^^
 
-The ``.ix`` indexer is deprecated, in favor of the more strict ``.iloc`` and ``.loc`` indexers. ``.ix`` offers a lot of magic on the inference of what the user wants to do. To wit, ``.ix`` can decide to index *positionally* OR via *labels*, depending on the data type of the index. This has caused quite a bit of user confusion over the years. The full indexing documentation is :ref:`here <indexing>`. (:issue:`14218`)
```
Comment (dsaxton) [inline doc/source/whatsnew/v0.20.0.rst:1311]: Same as above

Comment (asharma13524) [reply doc/source/whatsnew/v0.20.0.rst:1311]: so should I change back to the original "wit" in both instances and resubmit the pull request? Thanks.

**Review by dsaxton — APPROVED** [review]
Thanks @asharma13524

File: doc/source/whatsnew/v0.20.0.rst (hunk 2 above)
Comment (dsaxton) [reply doc/source/whatsnew/v0.20.0.rst]: You can simply change it back and commit, which it looks like you've done

**Comment by MarcoGorelli** [comment]
Thanks @asharma13524 !

**Pull request merged**